src/main/java/net/snowflake/client/jdbc/SnowflakeChunkDownloader.java

```diff
@@ -636,15 +636,16 @@ public SnowflakeResultChunk getNextChunkToConsume()
   private void waitForChunkReady(SnowflakeResultChunk currentChunk) throws InterruptedException {
     int retry = 0;
     long startTime = System.currentTimeMillis();
-    while (currentChunk.getDownloadState() != DownloadState.SUCCESS && retry < maxHttpRetries) {
+    while (true) {
       logger.debug(
           "Thread {} is waiting for #chunk{} to be ready, current" + "chunk state is: {}, retry={}",
           Thread.currentThread().getId(),
           nextChunkToConsume,
           currentChunk.getDownloadState(),
           retry);
 
-      if (currentChunk.getDownloadState() != DownloadState.FAILURE) {
+      if (currentChunk.getDownloadState() != DownloadState.FAILURE
+          && currentChunk.getDownloadState() != DownloadState.SUCCESS) {
         // if the state is not failure, we should keep waiting; otherwise, we skip
```

Comment:
Will update the comment here in a later PR. We don't wait if state is failure or success, only wait if in progress or not started.